packages/lib-panoptes-js/src/panoptes.js

```diff
@@ -52,22 +53,49 @@ function getQueryParams (query) {
   }
 }
 
+/*
+Converts a query object (e.g. { foo: 'bar', img: 'http://foo.bar/baz.jpg' })
+into a query string (e.g. "foo=bar&img=http%3A%2F%2Ffoo.bar%")
+ */
+function getQueryString (query, endpoint = '') {
+  const queryParams = getQueryParams(query)
+  let queryString = Object.entries(queryParams)
+    .filter(([key, val]) => val !== undefined)
+    .map(([key, val]) => `${encodeURIComponent(key)}=${encodeURIComponent(val)}`)
+    .join('&')
+  
+  if (queryString.length > 0) {
+    return endpoint.includes('?')
+      ? `&${queryString}`  // If endpoint already has a query '?', prepend the query string with a '&'.
+      : `?${queryString}`  //  Otherwise, prepend with a '?'.
+  }
+  return ''
+}
```

Comment:
You can use [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) to do this in all browsers and in Node too.

Reply:
```js
const searchParams = new URLSearchParams(query)
return searchParams.toString()
```